vllm/model_executor/layers/fused_moe/batched_deep_gemm_moe.py

```diff
@@ -101,11 +102,84 @@ def _silu_mul_fp8_quant_deep_gemm(
         t += 1
 
 
+@triton.jit
+def _silu_mul_fp8_quant_deep_gemm_ue8m0(
+    input_ptr,
+    y_q_ptr,
+    y_s_ptr,
+    counts_ptr,
+    H: tl.constexpr,
+    GROUP_SIZE: tl.constexpr,
+    stride_i_e,
+    stride_i_t,
+    stride_i_h,
+    stride_yq_e,
+    stride_yq_t,
+    stride_yq_h,
+    stride_ys_e,
+    stride_ys_t,
+    stride_ys_g,
+    stride_counts_e,
+    eps: tl.constexpr,
+    fp8_min: tl.constexpr,
+    fp8_max: tl.constexpr,
+    BLOCK: tl.constexpr,
+):
+    G = H // GROUP_SIZE
+
+    pid = tl.program_id(0)
+    e = pid // G
+    g = pid % G
+
+    e = e.to(tl.int64)
+    g = g.to(tl.int64)
+
+    n_tokens = tl.load(counts_ptr + e * stride_counts_e).to(tl.int64)
+
+    cols = tl.arange(0, BLOCK).to(tl.int64)
+    mask_h = cols < BLOCK
+
+    t = tl.zeros([], tl.int64)
+    while t < n_tokens:
+        base_i_offset = (e * stride_i_e + t * stride_i_t +
+                         g * GROUP_SIZE * stride_i_h)
+        base_yq_offset = (e * stride_yq_e + t * stride_yq_t +
+                          g * GROUP_SIZE * stride_yq_h)
+        base_ys_offset = e * stride_ys_e + t * stride_ys_t + g * stride_ys_g
+
+        x = tl.load(input_ptr + base_i_offset + cols * stride_i_h,
+                    mask=mask_h,
+                    other=0.0).to(tl.float32)
+        y2 = tl.load(input_ptr + base_i_offset + H * stride_i_h +
+                     cols * stride_i_h,
+                     mask=mask_h,
+                     other=0.0).to(tl.float32)
+
+        x = x * (1.0 / (1.0 + tl.exp(-x)))
+        y = x * y2
+
+        _absmax = tl.maximum(tl.max(tl.abs(y)), eps)
+        # round scale up to nearest power-of-two (UE8M0)
+        # specifically for DeepGemm SM100 kernels
+        scale_raw = _absmax / fp8_max
+        y_s = tl.math.exp2(tl.ceil(tl.log2(scale_raw)))
```

Comment:
Why can't you just add this as a conditional to the original kernel and pass in use_e8m0 as a constant bool i.e. `use_e8m0 : tl.constexpr`